improver_tests/nbhood/circular_kernel/test_CircularNeighbourhood.py

```diff
@@ -122,48 +117,47 @@ def test_multi_point_multitimes(self):
             expected[0][8 + index][8:13] = slice_
         for index, slice_ in enumerate(SINGLE_POINT_RANGE_3_CENTROID):
             expected[1][5 + index][5:10] = slice_
-        ranges = 3
-        result = CircularNeighbourhood(weighted_mode=True).apply_circular_kernel(
-            cube, ranges
-        )
+        radius = 6000
+        result = Neighbourhood("circular", weighted_mode=True).run(cube, radius)
         self.assertArrayAlmostEqual(result.data, expected)
 
     def test_single_point_masked_to_null(self):
-        """Test behaviour with a masked non-zero point. The behaviour here is
-        not right, as the mask is ignored. This comes directly from the
-        scipy.ndimage.correlate base behaviour."""
-
-        self.cube.data[7, 7] = 0
+        """Test behaviour with a masked non-zero point."""
 
-        expected = np.ones_like(self.cube.data)
+        self.cube.data[7, 7] = np.nan
         mask = np.zeros_like(self.cube.data)
         mask[7][7] = 1
         self.cube.data = np.ma.masked_array(self.cube.data, mask=mask)
-        for index, slice_ in enumerate(SINGLE_POINT_RANGE_3_CENTROID):
-            expected[5 + index][5:10] = slice_
-        ranges = 3
-        result = CircularNeighbourhood(weighted_mode=True).apply_circular_kernel(
-            self.cube, ranges
-        )
+        expected = np.ma.masked_array(self.cube.data, mask=mask)
+        radius = 6000
+        result = Neighbourhood("circular", weighted_mode=True).run(self.cube, radius)
         self.assertArrayAlmostEqual(result.data, expected)
+        self.assertArrayEqual(result.data.mask, expected.mask)
 
     def test_single_point_masked_other_point(self):
         """Test behaviour with a non-zero point next to a masked point.
-        The behaviour here is not right, as the mask is ignored."""
+        """
 
         self.cube.data[7, 7] = 0
 
         expected = np.ones_like(self.cube.data)
         mask = np.zeros_like(self.cube.data)
         mask[6][7] = 1
         self.cube.data = np.ma.masked_array(self.cube.data, mask=mask)
+
         for index, slice_ in enumerate(SINGLE_POINT_RANGE_3_CENTROID):
             expected[5 + index][5:10] = slice_
-        ranges = 3
-        result = CircularNeighbourhood(weighted_mode=True).apply_circular_kernel(
-            self.cube, ranges
-        )
+        # Update expected result for rows affected by masked point
+        expected[5:9, 4:11] = [
+            [1, 0.991736, 0.966102, 0.957265, 0.966102, 0.991736, 1],
+            [1, 0.966667, 0.940171, np.nan, 0.940171, 0.966667, 1],
+            [1, 0.958677, 0.932203, 0.923077, 0.932203, 0.958677, 1],
+            [1, 0.967741, 0.942149, 0.933333, 0.942149, 0.967741, 1],
+        ]
```

Comment:
I think that this is overwriting all but one row within `expected` compared to lines 148-149 meaning that a combination of these two methods of setting `expected` gives the intended result. In this PR, or later, it would be good to consolidate this.

Reply:
Done